### DISCUSSION TITLE
Using firebase real time database this happens => Your API key is invalid, please check you have copied it correctly.

### DISCUSSION THREAD

**Discussion by Jagadesh-Ram** [vercel/next.js#14420] [Help] [answered]
I have added .env, .env.local, .env.production to store all the necessary API keys...when accessed with process.env.keyname the following error pops, 
```
code: "auth/invalid-api-key"
message: "Your API key is invalid, please check you have copied it correctly."
```
but when directly the key is given, its works like charm!

**Comment by jamesmosier** [comment]
Are you trying to use this environment variable on the client side or server side? If you want to use it client side you must [prefix your variable](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser) with `NEXT_PUBLIC_` to ensure is exposed to the browser.

**Reply by Jagadesh-Ram** [reply]
Thanks for responding, am trying to sign in to the firebase using firebase.auth().signInAnonymously() in server side itself. 
Am using a method inside getStaticProps to do all this and fetch data...
Basically all i wanted to do was, get all the data to be populated on a page during the build time itself.

**Reply by jamesmosier** [reply]
Gotcha. Two questions:

1. What is the result when you do `console.log(process.env.YOUR_ENV_KEY)` in your `getStaticProps` method?
2. Do you have the package `dotenv` installed (because that would break this)?

**Reply by Jagadesh-Ram** [reply]
I did not install dotenv package separately, but its is available somehow! I used nextjs starter example repo.
Now I refracted my code to do sign in and fetching the data into a separate function and called that in my getStaticProps method. It also prints the correct  api keys and the sign in is also done perfectly but the data is not fetched :(

**Reply by seiwonpark** [reply]
@jamesmosier  Thank you so much!! in my `.env.local` file, I just used API_KEY not NEXT_PUBLIC_API_KEY. Thank you!

**Comment by Jagadesh-Ram** [comment, marked as answer]
Alright, i now fixed the firebase data fetching using a simple way(deprecated one TBH!).
No need to import any firebase sdk!!!
Just use this method, but for that one needs have the secret key and firebase database url(easy to get from config file in firebase console) to the particular real time database.
It can be viewed here 
`https://console.firebase.google.com/project/'projectname'/settings/serviceaccounts/databasesecrets 
`
copy the secret key and add them in .env file under the name `FIREBASE_SECRET` and **Never Ever Post it in a public repo**. Now use the below code to get the files as JSON from the database!!!
```
 export const getStaticProps = async (context) => {
  const fetchedData = await fetch(`${process.env.FIREBASE_DATABASE_URL}/filename.json?orderBy="$key"&limitToLast=50&auth=${process.env.FIREBASE_SECRET}`);
  const data = await fetchedData.json();
  return {
    props: {
      data,
    },
  };
};

```

**Comment by internetjonathan** [comment]
This is happening to me now. The extremely odd thing is that its been working all week and out of nowhere I could no longer log into my app.  When i tried through postman just to make sure it wasn't my front end it got your same error about the API being incorrect.  I've importing the config file like such from the beginning with no problem :
`const config = require('../util/config')
`
so now i just pasted the config variable into my file rather than importing it and it now works again.  I kid you not, for the past two weeks i have been importing that config file without any issues. i do not understand the change.

**Comment by dht7166** [comment]
Quick follow up for anyone using firebase. The api key are supposed to be facing client-side. This api key is not the admin SDK key.

More info here: https://stackoverflow.com/questions/37482366/is-it-safe-to-expose-firebase-apikey-to-the-public
